### DISCUSSION TITLE
Calculate resource usage using the Milvus sizing tool

### DISCUSSION THREAD

**Discussion by fz5400** [milvus-io/milvus#30541] [Q&A and General discussion] [answered]
Hi, I'm a Milvus newbie, and I have some confusion about using Milvus sizing tool,
I want to build a vector database of about 6 million 128-dimensional data, using the HNSW index.
<img width="418" alt="截屏2024-02-05 17 55 43" src="https://github.com/milvus-io/milvus/assets/89795046/e6a384b8-3bb6-476b-a68b-914457ad867c">

Approximate capacity is as follows：
<img width="845" alt="截屏2024-02-05 17 55 52" src="https://github.com/milvus-io/milvus/assets/89795046/b41a4ebb-1b12-4f39-b825-f4b68d9f475a">

What confuses me is that the minimum Milvus cluster setup seems to require a lot of resources to build, and doesn't match the memory usage above. According to this resource configuration, I probably need a 42.5-core CPU and 175.5GB of memory to build a cluster.
<img width="812" alt="截屏2024-02-05 17 56 12" src="https://github.com/milvus-io/milvus/assets/89795046/3374a8ad-f550-45af-abee-ffaa801a8e74">

**Comment by xiaofan-luan** [comment]
what is the problem?
I thought you pick the wrong dimension

**Reply by fz5400** [reply]
sorry about the wrong dimension problem, I have fix that below. However, the resource usage of the Minimum Milvus cluster setup still greatly exceeded my expectations.
<img width="1306" alt="截屏2024-02-06 09 46 03" src="https://github.com/milvus-io/milvus/assets/89795046/6365eaf6-8b98-481b-a8be-2794a4756311">
<img width="851" alt="截屏2024-02-06 09 46 12" src="https://github.com/milvus-io/milvus/assets/89795046/503a2cf0-bf09-42a9-b07e-9d16a9252f79">

**Microservice components**
CPU cores:
1 (Root Coord) + 0.5 (Index Coord) + 0.5 (Query Coord) + 0.5 (Data Coord) + 1 (Proxy) + 3 (Query Node) + 1 (Data Node) + 4 (Index Node) =  **11.5 cores**

Memory:
2 (Root Coord) + 1.5 (Index Coord, Query Coord, Data Coord) + 4 (Proxy) + 12 (Query Node) + 4 (Data Node) + 8 (Index Node) = **31.5 GB**

**Third-party dependencies**
CPU cores:
2 *  3 (ETCD) + 2 * 4 (Minio) + 2 * 3 (Pulsar - Bookie) + 2 * 2 (Pulsar - Broker) + 2 * 2  (Pulsar - Proxy) + 1* 3  (Pulsar -Zookeeper) = **31 cores**

Memory:
4 * 3 (ETCD) + 8 * 4 (Minio) + 16 * 3 (Pulsar - Bookie) + 18 * 2 (Pulsar - Broker) + 5 *2 (Pulsar - Proxy) + 2 * 3 (Pulsar -Zookeeper) = **144 GB**

To sum up, 42.5-core CPU and 175.5GB of memory will be needed to build a cluster. I just want to store 5 million pieces of data. Are so many resources really necessary?

**Reply by yhmo** [reply, marked as answer]
It is a recommendation. The tool estimates the resources requirement based on our experience, a safety ratio(2.0 ~ 3.0) is multiply.

**Reply by fz5400** [reply]
Thank you very much for your answers. That means that if I want to deploy the milvus vector database in a production environment, it is best to follow the recommended resource allocation, if I understand it correctly.

**Reply by yhmo** [reply]
Yes, it is better to follow the recommendation.
